pkg/apis/config/retention.go

```diff
@@ -45,26 +131,42 @@ func (cfg *RetentionPolicy) Equals(other *RetentionPolicy) bool {
 	}
 
 	return other.RunAt == cfg.RunAt &&
-		other.MaxRetention == cfg.MaxRetention
+		other.DefaultRetention == cfg.DefaultRetention
 }
 
 func newRetentionPolicyFromMap(cfgMap map[string]string) (*RetentionPolicy, error) {
 	rp := RetentionPolicy{
-		RunAt:        DefaultRunAt,
-		MaxRetention: DefaultMaxRetention,
+		RunAt:            DefaultRunAt,
+		DefaultRetention: DefaultDefaultRetention,
 	}
 
 	if schedule, ok := cfgMap[runAt]; ok {
 		rp.RunAt = schedule
 	}
 
-	if duration, ok := cfgMap[maxRetention]; ok {
-		v, err := strconv.Atoi(duration)
+	if duration, ok := cfgMap[defaultRetention]; ok {
+		v, err := ParseDuration(duration)
+		if err != nil {
+			return nil, fmt.Errorf("incorrect configuration for defaultRetention: %w", err)
+		}
+		rp.DefaultRetention = v
+	} else if duration, ok := cfgMap[maxRetention]; ok {
+		// Deprecated: maxRetention is deprecated, use defaultRetention instead.
```

Comment:
I wish we could log a warning here but it looks like it might be quite a refactor to get a context into this method

Reply:
I did that using standard log. We don't pass logger to config package, so I had to use that.